src/osd/modules/lib/osdlib.h

```diff
@@ -48,14 +53,53 @@ void osd_process_kill(void);
 
 int osd_setenv(const char *name, const char *value, int overwrite);
 
+
 /*-----------------------------------------------------------------------------
     osd_get_clipboard_text: retrieves text from the clipboard
 
     Return value:
 
         the returned string needs to be osd_free()-ed!
-
 -----------------------------------------------------------------------------*/
+
 char *osd_get_clipboard_text(void);
 
+
+/*-----------------------------------------------------------------------------
+    osd_dynamic_bind: load functions from optional shared libraries
+
+    Notes:
+
+        - Supports Mac OS X, Unix and Windows (both desktop and Windows
+          Store universal applications)
+        - A symbol can be searched in a list of libraries (e.g. more
+          revisions of a same library)
+-----------------------------------------------------------------------------*/
+
+class osd_dynamic_bind_base
+{
+public:
+	osd_dynamic_bind_base(const char *symbol, const std::vector<std::wstring> libraries);
+	~osd_dynamic_bind_base();
+
+protected:
+	void *m_module;
```

Comment:
Having a reference to the module in every dynamically bound function is inefficient - it would be better to have a class osd_dynamic_module that holds a reference to the module and exposes a template method for looking up a symbol.